tests/base_scraper/test_base_scraper.py

```diff
@@ -5,133 +5,120 @@
-
-    def test_destination(self):
-        """Test for various destination scenarios"""
-
-        filename = 'download_test.txt'
-        test_url = urljoin(self.wdir, filename)
-
-        # destination is directory
-        scraper = mozdownload.DirectScraper(url=test_url,
-                                            destination=self.temp_dir,
-                                            logger=self.logger)
-        self.assertEqual(scraper.filename, os.path.join(self.temp_dir, filename))
-
-        # destination has directory path with filename
-        destination = os.path.join(self.temp_dir, filename)
-        scraper = mozdownload.DirectScraper(url=test_url,
-                                            destination=destination,
-                                            logger=self.logger)
-        self.assertEqual(scraper.filename, destination)
-
-        # destination only has filename
-        scraper = mozdownload.DirectScraper(url=test_url,
-                                            destination=filename,
-                                            logger=self.logger)
-        self.assertEqual(scraper.filename, os.path.abspath(filename))
-
-        # destination directory does not exist
-        destination = os.path.join(self.temp_dir, 'temp_folder', filename)
-        scraper = mozdownload.DirectScraper(url=test_url,
-                                            destination=destination,
-                                            logger=self.logger)
-        self.assertEqual(scraper.destination, destination)
-
-        # ensure that multiple non existing directories are created
-        destination = os.path.join(self.temp_dir, 'tmp1', 'tmp2', filename)
-        scraper = mozdownload.DirectScraper(url=test_url,
-                                            destination=destination,
-                                            logger=self.logger)
-        self.assertEqual(scraper.destination, destination)
+
+@pytest.mark.skip(reason="Permanent failure due to mozqa.com not available anymore (#492)")
+def test_valid_authentication(tmpdir):
+    """testing with valid authentication"""
+    username = 'mozilla'
+    password = 'mozilla'
+    basic_auth_url = 'http://mozqa.com/data/mozqa.com/http_auth/basic/'
+    scraper = mozdownload.DirectScraper(destination=str(tmpdir),
+                                        url=basic_auth_url,
+                                        username=username,
+                                        password=password)
+    scraper.download()
+    assert os.path.isfile(os.path.join(str(tmpdir), 'mozqa.com'))
+
+def test_destination_isdirectory(httpd, tmpdir):
+    """destination is directory"""
+    filename = 'download_test.txt'
+    test_url = urljoin(httpd.get_url(), filename)
+    scraper = mozdownload.DirectScraper(url=test_url, destination=str(tmpdir))
+    assert scraper.filename == os.path.join(str(tmpdir), filename)
+
+
+def test_destination_hasdirectory(httpd, tmpdir):
```

Comment:
You stripped an important part of the former comment. This is a directory with a filename. Also having a good method name would make the doc comment obsolete. So best would be to use the content of the former comments as method name.

Reply:
It was added as a doc string in the function itself. I will rename the function as you asked. Please let me know if i missed anything.

Reply:
That looks better. Thanks!